### PR TITLE
[WIP] CI Experiments

### PR DESCRIPTION
**Pull request by bryevdv**

### CODE REVIEW

File: .github/workflows/bokeh-ci.yml [hunk 1]
```diff
@@ -4,38 +4,46 @@ on: [pull_request]
...
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        python_version: [3.7]
-        os: [windows-latest, macos-latest]
+    runs-on: ubuntu-latest
+    env:
+      CONDA_REQS: "conda=4.7.12 conda-build=3.18.10 conda-verify=3.4.2 ripgrep=0.10.0"
```
Comment (mattpap) [inline .github/workflows/bokeh-ci.yml]: btw. What's the purpose for `ripgrep` in the build? It fails to install on Windows (at least 32-bit) and isn't mentioned anywhere explicitly except for `meta.yaml`.

Comment (bryevdv) [reply .github/workflows/bokeh-ci.yml]: supposedly an optional dep that makes some aspect of conda quite faster.

**Comment by bryevdv** [comment]
@mattpap  any idea what could cause this to fail:

```
    
>       assert ext.build(tmp) is True
E       AssertionError: assert False is True
E        +  where False = <function build at 0x7f34c8780170>('/tmp/pytest-of-runner/pytest-0/test_ext_commands0/bk_ext_01')
E        +    where <function build at 0x7f34c8780170> = ext.build
```
It's the only test preventing osx and linux tests from passing using a unified pipeline (and windows too, quite close)

**Comment by mattpap** [comment]
> any idea what could cause this to fail:

This is not the entire failure; there's console output below that. TypeScript's stdlib is missing (bokeh/server/static/lib).

**Comment by bryevdv** [comment]
@mattpap JS tests are failing on CI with this, and also for me locally, is there some new thing that needs to be installed or done to run js unit tests?
```
[19:28:49] Starting 'test:unit'...
INTERNAL ERROR: { Error: connect ECONNREFUSED 127.0.0.1:9222
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1113:14)
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 9222 }
[19:28:51] failed: Error: tests failed
[19:28:51]     at ChildProcess.proc.on (/Users/bryan/work/bokeh/bokehjs/make/tasks/test.ts:99:16)
[19:28:51]     at ChildProcess.emit (events.js:182:13)
[19:28:51]     at ChildProcess.EventEmitter.emit (domain.js:442:20)
[19:28:51]     at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
```

**Comment by mattpap** [comment]
> JS tests are failing on CI with this, and also for me locally, is there some new thing that needs to be installed or done to run js unit tests?

For unit tests yes, but in general no (when running `node make test`). Integration tests have depended on headless chrome for a while, now all tests depend on it.

File: .github/workflows/bokeh-ci.yml [hunk 2]
```diff
@@ -262,17 +262,19 @@ jobs:
...
+          sudo apt-get update
+          sudo apt-get install software-properties-common
+          sudo add-apt-repository ppa:canonical-chromium-builds/stage
+          sudo apt-get update
+          sudo apt-get install chromium-browser
+
+      - name: Start chrome headless
+        shell: bash
+        run: |
+          chromium-browser --headless --no-sandbox --remote-debugging-port=9222
```
Comment (mattpap) [inline .github/workflows/bokeh-ci.yml]: `--no-sandbox` was a requirement for Travis CI. I'm not sure, but it may not be necessary here.

Comment (bryevdv) [reply .github/workflows/bokeh-ci.yml]: 👍  I'll give it a try without at some point

**Comment by mattpap** [comment]
I noticed there is no color output currently. Maybe we should run `pytest` with `--color=yes`, or use some other approach.

**Comment by bryevdv** [comment]
@ammarnajjar I am trying to get a consistent, consolidated workflow across all OSes. Things are *almost* working but I can't get windows unit tests to run. There  is always an import error on numpy DLLs. Do you have any ideas here?

**Comment by bryevdv** [comment]
This looks useful (maybe vendor): https://github.com/s-weigand/setup-conda

**Comment by bryevdv** [comment]
@mattpap FYI this currently reproduces all the existing CI jobs (and more). There is an issue with some of the selenium tests which I expect is more to do with the test. I will either add/increase flaky tolerances on some of them, or mark them `skip` for now with a follow-up issue. I do intend to merge this by tomorrow.

**Comment by bryevdv** [comment]
OK this is now green, with all previous jobs covered, so I am going to go ahead and merge. There will need to be some follow on work:

* cleanup/remove crufts from other CI systems
* some integration tests are skipped for now
* cleanup/factoring of ci.yml
* examples tests improvements:

  add a task to upload or report upload location, so no scrolling needed to get to report

  job is 3-4x longer than almost every other job. Splitting up would greatly increase test throughput.

cc @mattpap

**Pull request merged**